### DISCUSSION TITLE
Typescript x Signals

### DISCUSSION THREAD

**Discussion by Delmotte-Vincent** [preactjs/preact#4215] [General]
Hi, I was searching for using signals with Typescript but I don't find any information about that. Is it something that is plan to do or signal will never support typescript ?

**Comment by marvinhagemeister** [comment]
They already support TypeScript out of the box. Is there something that you feel is missing from the experience?

**Reply by Delmotte-Vincent** [reply]
Is there documentation about that ?

**Reply by marvinhagemeister** [reply]
I don't understand the question. It works exactly the same way in JavaScript vs TypeScript. The only difference is that you can pass some generics in case the type cannot be auto inferred.

```jsx
interface MyComplexType {
  foo: number;
  bar: string | null;
}

const sig = signal<MyComplexType | null>(null)

// same thing for the hook variant
const sig = useSignal<MyComplexType | null>(null)
```

That would conclude the whole documentation for TypeScript. I wouldn't know what else to mention in a TypeScript specific documentation. That's really all there is to it.

**Reply by Delmotte-Vincent** [reply]
Thanks a lot, that was my question, I wish you a nice day !
